### PR TITLE
Add `SSEEventID` to validate the id received from the server.

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

File: libparsec/crates/client/src/connection_monitor.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 use std::{ops::ControlFlow, sync::Arc};
 
 use libparsec_client_connection::{
-    AuthenticatedCmds, ConnectionError, RateLimiter, SSEResponseOrMissedEvents,
+    AuthenticatedCmds, ConnectionError, RateLimiter, SSEId, SSEResponseOrMissedEvents,
```
Comment (touilleMan) [inline libparsec/crates/client/src/connection_monitor.rs:6]: Call it `SSEEventID` ? given it is closely related to `SSEEvent`

(and with capital `ID` given that's what we do in `VlobID`, `EntryID` etc.)

Comment (FirelightFlagboy) [reply libparsec/crates/client/src/connection_monitor.rs]: done

File: libparsec/crates/client_connection/src/authenticated_cmds/mod.rs [hunk 2]
```diff
@@ -228,7 +228,7 @@ impl AuthenticatedCmds {
         log::trace!("Will listen for SSE event at {}", response.url());
         let mut sse_stream = response.bytes_stream().eventsource();
         if let Some(last_event_id) = last_event_id {
-            sse_stream.set_last_event_id(last_event_id);
+            sse_stream.set_last_event_id(last_event_id.to_str().expect("last-event-id first originated from a String so converting it back should not cause problem"));
```
Comment (TimeEngineer) [inline libparsec/crates/client_connection/src/authenticated_cmds/mod.rs:231]: make a method instead, so we can avoid explaining next time

Comment (FirelightFlagboy) [reply libparsec/crates/client_connection/src/authenticated_cmds/mod.rs]: done

Comment (mmmarcos) [reply libparsec/crates/client_connection/src/authenticated_cmds/mod.rs]: I do not see this change applied, did I missed something? :thinking:

File: libparsec/crates/client_connection/src/error.rs [hunk 3]
```diff
@@ -65,6 +65,9 @@ pub enum ConnectionError {
     /// We failed to deserialize ApiVersion
     #[error("Wrong ApiVersion {0}")]
     WrongApiVersion(String),
+
+    #[error("Invalid sse id: {0}")]
+    InvalidSSEId(#[from] reqwest::header::InvalidHeaderValue),
```
Comment (mmmarcos) [inline libparsec/crates/client_connection/src/error.rs:70]: Also rename here?
```suggestion
    #[error("Invalid sse event id: {0}")]
    InvalidSSEEventId(#[from] reqwest::header::InvalidHeaderValue),
```

File: libparsec/crates/client_connection/src/authenticated_cmds/mod.rs (hunk 2 above)
Comment (mmmarcos) [reply libparsec/crates/client_connection/src/authenticated_cmds/mod.rs]: Oh it was moved, sorry :sweat_smile:

File: libparsec/crates/client_connection/src/error.rs (hunk 3 above)
Comment (FirelightFlagboy) [reply libparsec/crates/client_connection/src/error.rs]: Indeed, done

**Review by TimeEngineer — APPROVED** [review]

**Pull request merged**